### PR TITLE
Fix some findings

### PR DESCRIPTION
**Pull request by katietz**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/build_mamba.bat [hunk 1]
```diff
@@ -10,6 +10,8 @@ rmdir /Q /S build
 mkdir build
 cd build
 
+set "CXXFLAGS=%CXXFLAGS% /D_LIBCPP_DISABLE_AVAILABILITY=1"
```
Comment (wolfv) [inline recipe/build_mamba.bat:13]: I think this is only relevant on macOS, isn't it?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -46,7 +47,7 @@ outputs:
         - termcolor-cpp
         - cli11
         - fmt
-        - libiconv                               # [unix]
+        - libiconv                               # [osx]
```
Comment (wolfv) [inline recipe/meta.yaml:50]: Maybe we can also get rid of `libiconv` here? I think it's a dependency of `libarchive` and we don't link with it directly.

File: recipe/build_mamba.bat (hunk 1 above)
Comment (katietz) [reply recipe/build_mamba.bat:13]: well, can be. it doesn't hurt.  any on unix we define it unconditionally anyway too.  but sure, we can remove

File: recipe/meta.yaml (hunk 2 above)
Comment (katietz) [reply recipe/meta.yaml:50]: right, I wasn't brave enough to remove it completely.  But my tests have shown that there is neither a shared linking happening, nor a statically one

Comment (katietz) [reply recipe/meta.yaml]: done

**Comment by katietz** [comment]
@conda-forge-admin, please rerender

**Comment by hmaarrfk** [comment]
Sorry i might have merged something else that would create a conflict. is this good to go otherwise?

**Pull request closed without merging**